factoryWithTypeCheckers.js

```diff
@@ -364,6 +365,36 @@ module.exports = function(isValidElement, throwOnDirectAccess) {
     return createChainableTypeChecker(validate);
   }
 
+
+  function createElementWithTypeChecker(expectedType) {
+    var ACCEPTABLE_TYPES_OF_EXPECTED_TYPES = ['string', 'function'];
+
+    if (ACCEPTABLE_TYPES_OF_EXPECTED_TYPES.indexOf(typeof expectedType) === -1) {
+      process.env.NODE_ENV !== 'production' ? warning(false, 'Invalid argument supplied to ElementWithType, expected an Html tag name or a Component.') : void 0;
+      return emptyFunction.thatReturnsNull;
+    }
+
+    function getDisplayName (comp) {
+      return typeof comp === 'function' ? comp.displayName || comp.name : comp;
+    }
+
+    function validate(props, propName, componentName, location, propFullName) {
+      var propValues = [].concat(props[propName]);
```

Comment:
this should just be `props[propName]`; there's no reason to support multiple values here.

Reply:
Are you saying the support of multiple element types should be using `oneOf`?

Reply:
Yes.